api/src/main/java/org/openmrs/api/impl/ConditionServiceImpl.java

```diff
@@ -101,46 +103,53 @@ public List<Condition> getConditionsByEncounter(Encounter encounter) throws APIE
 	 */
 	@Override
 	public Condition saveCondition(Condition condition) throws APIException {
-		Condition existingCondition = Context.getConditionService().getConditionByUuid(condition.getUuid());
+		
 		// If there is no existing condition, then we are creating a condition
-		if (existingCondition == null) {
+		Integer existingConditionId = condition.getConditionId();
+		if (existingConditionId == null) {
 			return conditionDAO.saveCondition(condition);
 		}
-
-		// If the incoming condition has been voided, we simply void the existing condition
-		// All other changes are ignored
-		if (condition.getVoided()) {
-			if (!existingCondition.getVoided()) {
-				return Context.getConditionService().voidCondition(existingCondition,
-					StringUtils.isNotBlank(condition.getVoidReason()) ? condition.getVoidReason() : "Condition deleted");
-			} else {
-				return existingCondition;
-			}
+		
+		// If there is an existing condition, create a new condition from it and reset the existing instance state
+		Condition newCondition = Condition.newInstance(condition);
+		Context.refreshEntity(condition);
+		
+		// Determine how the existing state and new state have changed
+		boolean conditionHasChanged = !newCondition.matches(condition);
+		boolean existingVoided = BooleanUtils.isTrue(condition.getVoided());
+		boolean newVoided = BooleanUtils.isTrue(newCondition.getVoided());
+		boolean unVoidOriginal = existingVoided && !newVoided;
```

Comment:
we unvoid the original even if the new one differs from the old? wouldn't we just create a new condition in this case that we both want to unvoid and modify?

Reply:
It's a good point.  I had originally re-written that way, and it was failing, which is why I preserved the existing behavior.  I'll have a look.

Reply:
OK, per @mogoodrich comment and suggestion, I've removed the special "unvoid" handling, which I agree is unnecessary and wrong.  Unvoiding shouldn't eliminate the audit trail of the previously voided data, it should just be treated like an edit, linking back to voided row.  That is now working.